e2e/tests/controls/polygon-controls/index.spec.ts

```diff
@@ -1,83 +1,87 @@
-    object.strokeLineJoin = 'miter';
-    object.strokeWidth = 30;
-    object.exactBoundingBox = true;
-    object.setDimensions();
-    object.set('dirty', true);
-    object.canvas.renderAll();
-  });
+    const modifyPoly = (
+      title: string,
+      controlName: string,
+      delta: { x: number; y: number }
+    ) =>
+      test.step(title, async () => {
+        const eventStreamHandle =
+          await test.step('subscribe to poly events', () =>
+            starUtil.evaluateHandle((object) => {
+              const events: { action: string; corner: string }[] = [];
+              const disposer = object.on(
+                'modifyPoly',
+                ({ transform: { action, corner } }) =>
+                  events.push({ action, corner })
+              );
+              object.once('modified', ({ transform: { action, corner } }) => {
+                disposer();
+                events.push({ action, corner });
+              });
+              return events;
+            }));
 
-  expect(await canvasUtil.screenshot()).toMatchSnapshot({
-    name: 'initial-exact.png',
-  });
+        const controlPoint = await starUtil.getObjectControlPoint(controlName);
+        await page.mouse.move(controlPoint.x, controlPoint.y);
+        await page.mouse.down();
+        await page.mouse.move(
+          controlPoint.x + delta.x,
+          controlPoint.y + delta.y,
+          {
+            steps: 40,
+          }
+        );
+        await page.mouse.up();
 
-  const p2ControlPoint = await starUtil.getObjectControlPoint('p2');
+        expect(await canvasUtil.screenshot()).toMatchSnapshot({
+          name: toSnapshotName(`moved_controls_${controlName}`, exact),
+        });
 
-  // drag the p2 control
-  await page.mouse.move(p2ControlPoint.x, p2ControlPoint.y);
-  await page.mouse.down();
-  await page.mouse.move(p2ControlPoint.x + 300, p2ControlPoint.y + 100, {
-    steps: 40,
-  });
-  await page.mouse.up();
-  expect(await canvasUtil.screenshot()).toMatchSnapshot({
-    name: 'moved_controls_p2_exact.png',
-  });
+        const events = await eventStreamHandle.jsonValue();
+        expect(events.length).toBeGreaterThan(0);
+        expect(events).toMatchObject(
+          new Array(events.length).fill({
+            action: 'modifyPoly',
+            corner: controlName,
+          })
+        );
+      });
 
-  // drag in the opposite direction
-  const p4ControlPoint = await starUtil.getObjectControlPoint('p4');
-  await page.mouse.move(p4ControlPoint.x, p4ControlPoint.y);
-  await page.mouse.down();
-  await page.mouse.move(p4ControlPoint.x - 300, p4ControlPoint.y - 100, {
-    steps: 40,
-  });
-  await page.mouse.up();
-  expect(await canvasUtil.screenshot()).toMatchSnapshot({
-    name: 'moved_controls_p4_exact.png',
+    await modifyPoly('drag the p2 control', 'p2', { x: 300, y: 100 });
+    await modifyPoly('drag in the p4 control in the opposite direction', 'p4', {
+      x: -300,
+      y: -100,
+    });
   });
-});
+}
```

Comment:
this test refactor blocks me.
Can we leave the tests as a simple sequence of steps?
In the moment we want to do something different this change becomes a difficulty more than an helper.

We can verify the event firing with a simple unit test i believe, we don't need to add it here.